python/__init__.py.in

```diff
@@ -0,0 +1,6 @@
+import sys
+
+@F3D_PYTHON_FIXUP@
```

Comment:
This line makes the file not valid python anymore, which would get in the way of running linters/formatters (not a big deal now, but would be useful if/when more code ends up in there).

```suggestion
# @F3D_PYTHON_FIXUP@
```
making the line a comment would solve that (just need to make sure to output a dummy/comment first line on the cmake side)

---

Now to get really subjectively nitpicky, I'm not a fan of how much python the fixup injects, first because you've got python _logic_ tucked away in a cmake file, but also because you end up writing python _code_ as strings in cmake.

How about only generating data, in this case comma-separated python strings (with a comment line first, see first point), rather than full blown logic code?

```suggestion
F3D_ABSOLUTE_DLLS = [
    # @F3D_ABSOLUTE_DLLS_FIXUP@
]
F3D_RELATIVE_DLLS = [
    # @F3D_RELATIVE_DLLS_FIXUP@
]

if sys.version_info >= (3, 8) and sys.platform == "win32":
    for abs_path in F3D_ABSOLUTE_DLLS:
        os.add_dll_directory(abs_path)
    root = Path(__file__).parent
    for rel_path in F3D_RELATIVE_DLLS:
        os.add_dll_directory((root / rel_path).resolve())
```
That would make the `.py.in` a syntactically valid and logically complete version of the final `.py` file, just missing the injected data values. 
(also I believe you can import `os` and `pathlib.Path` at the top, they're never not going to be there and it shouldn't have significant performance impact)

The cmake part would have more cmake boilerplate but less python code (untested, no idea what i'm doing on this end):
```cmake
	if(WIN32)
		set(_fixup "generated at build\n") # first line will be a python comment because we prefixed with '# ' in .py.in
		foreach(_path ${F3D_DLL_FIXUP_ABSOLUTE_PATHS})
			set(_fixup "${_fixup}    '${_path}',\n")
		endforeach()
		set(${F3D_ABSOLUTE_DLL_FIXUP_OUTPUT} ${_fixup} PARENT_SCOPE)

		set(_fixup "generated at build\n") # python comment
		foreach(_path ${F3D_DLL_FIXUP_RELATIVE_PATHS})
			set(_fixup "${_fixup}    '${_path}',\n")
		endforeach()
		set(${F3D_RELATIVE_DLL_FIXUP_OUTPUT} ${_fixup} PARENT_SCOPE)
	else()
		set(${F3D_ABSOLUTE_DLL_FIXUP_OUTPUT} "empty because not build on/for windows" PARENT_SCOPE) # python comment
		set(${F3D_RELATIVE_DLL_FIXUP_OUTPUT} "empty because not build on/for windows" PARENT_SCOPE) # python comment
	endif()
```